fluent-plugin-events/lib/fluent/plugin/in_events.rb

```diff
@@ -32,11 +32,17 @@ class EventsInput < Fluent::Plugin::Input
       config_param :namespace, :string, default: nil
       config_param :label_selector, :string, default: nil
       config_param :field_selector, :string, default: nil
+      config_param :type_selector, :array, default: ["ADDED", "MODIFIED"], value_type: :string
       config_param :configmap_update_interval_seconds, :integer, default: 10
       config_param :watch_interval_seconds, :integer, default: 300
       
       def configure(conf)
         super
+
+        @valid_types = ["ADDED", "MODIFIED", "DELETED"]
+        raise Fluent::ConfigError, 'type_selector needs to be an array with maximum 3 elements: ADDED, MODIFIED and DELETED.' \
+          if @type_selector.length > 3 || !@type_selector.any? || !@type_selector.all? {|type| @valid_types.any? {|valid| valid.casecmp(type) == 0}}
```

Comment:
I think since we only care about exact equality (0 case) then we can use `casecmp?(other_str)` which returns true for equality and false for other cases

Reply:
oh, did not see that one